entities/src/main/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategy.kt

```diff
@@ -0,0 +1,62 @@
+package org.odk.collect.entities.javarosa.filter
+
+import org.javarosa.core.model.CompareToNodeExpression
+import org.javarosa.core.model.condition.EvaluationContext
+import org.javarosa.core.model.condition.FilterStrategy
+import org.javarosa.core.model.instance.DataInstance
+import org.javarosa.core.model.instance.TreeReference
+import org.javarosa.xpath.expr.XPathEqExpr
+import org.javarosa.xpath.expr.XPathExpression
+import org.odk.collect.entities.javarosa.intance.LocalEntitiesInstanceAdapter
+import org.odk.collect.entities.storage.EntitiesRepository
+import java.util.function.Supplier
+
+class LocalEntitiesFilterStrategy(entitiesRepository: EntitiesRepository) :
+    FilterStrategy {
+
+    private val dataAdapter = LocalEntitiesInstanceAdapter(entitiesRepository)
+
+    override fun filter(
+        sourceInstance: DataInstance<*>,
+        nodeSet: TreeReference,
+        predicate: XPathExpression,
+        children: MutableList<TreeReference>,
+        evaluationContext: EvaluationContext,
+        next: Supplier<MutableList<TreeReference>>
+    ): List<TreeReference> {
+        if (!dataAdapter.supportsInstance(sourceInstance.instanceId)) {
+            return next.get()
+        }
+
+        val candidate = CompareToNodeExpression.parse(predicate)
+        return when (val original = candidate?.original) {
+            is XPathEqExpr -> {
+                if (original.isEqual) {
```

Comment:
Could you explain why we need this only for this particular expression type? Is everything fine with how other expressions work?

Reply:
We only support equals (`=`) for now, and other expressions will just fall through to JavaRosa's standard set of `FilterStrategy` (the `next.get()` calls). In the future, we'll probably add support for more kinds of expressions though.